### PR TITLE
bags: Rename kinds with a more consistent naming scheme

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

**Comment by ajreynol** [comment]
@aniemetz and I were discussing: we think we may want to use `inter` instead of `intersection` for sets, to match the SMT-LIB standard `re.inter` for regular expression intersection.

Thus, I think I'd recommend `inter_min` instead of `intersection_min`.

File: src/theory/bags/normal_form.cpp [hunk 1]
```diff
@@ -458,14 +458,14 @@ Node NormalForm::evaluateIntersectionMin(TNode n)
...
   // Example
   // -------
   // input: (difference_subtract A B)
-  //    where A = (union_disjoint (MK_BAG "x" 4) (MK_BAG "z" 2)))
-  //          B = (union_disjoint (MK_BAG "x" 3) (MK_BAG "y" 1)))
+  //    where A = (union_disjoint (BAG_MAKE "x" 4) (BAG_MAKE "z" 2)))
+  //          B = (union_disjoint (BAG_MAKE "x" 3) (BAG_MAKE "y" 1)))
   // output:
-  //    (union_disjoint (MK_BAG "x" 1) (MK_BAG "z" 2))
+  //    (union_disjoint (BAG_MAKE "x" 1) (BAG_MAKE "z" 2))
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:468]: Ditto

File: src/theory/bags/normal_form.cpp [hunk 2]
```diff
@@ -579,8 +579,8 @@ Node NormalForm::evaluateCard(TNode n)
   // Examples
   // --------
   //  - (card (emptyBag String)) = 0
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:581]:
```suggestion
  //  - (card (bag.empty String)) = 0
```

File: src/theory/bags/normal_form.cpp [hunk 3]
```diff
@@ -579,8 +579,8 @@ Node NormalForm::evaluateCard(TNode n)
...
-  //  - (choose (MK_BAG "x" 4)) = 4
-  //  - (choose (union_disjoint (MK_BAG "x" 4) (MK_BAG "y" 1))) = 5
+  //  - (choose (BAG_MAKE "x" 4)) = 4
+  //  - (choose (union_disjoint (BAG_MAKE "x" 4) (BAG_MAKE "y" 1))) = 5
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:583]:
```suggestion
  //  - (bag.choose (bag.union_disjoint (bag "x" 4) (bag "y" 1))) = 5
```

File: src/theory/bags/normal_form.cpp [hunk 4]
```diff
@@ -600,11 +600,12 @@ Node NormalForm::evaluateIsSingleton(TNode n)
   // Examples
   // --------
   // - (bag.is_singleton (emptyBag String)) = false
-  // - (bag.is_singleton (MK_BAG "x" 1)) = true
-  // - (bag.is_singleton (MK_BAG "x" 4)) = false
-  // - (bag.is_singleton (union_disjoint (MK_BAG "x" 1) (MK_BAG "y" 1))) = false
+  // - (bag.is_singleton (BAG_MAKE "x" 1)) = true
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:603]:
```suggestion
  // - (bag.is_singleton (bag "x" 1)) = true
```

File: src/theory/bags/normal_form.cpp [hunk 5]
```diff
@@ -600,11 +600,12 @@ Node NormalForm::evaluateIsSingleton(TNode n)
...
+  // - (bag.is_singleton (BAG_MAKE "x" 4)) = false
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:604]:
```suggestion
  // - (bag.is_singleton (bag "x" 4)) = false
```

File: src/theory/bags/normal_form.cpp [hunk 6]
```diff
@@ -600,11 +600,12 @@ Node NormalForm::evaluateIsSingleton(TNode n)
   // Examples
   // --------
   // - (bag.is_singleton (emptyBag String)) = false
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:602]:
```suggestion
  // - (bag.is_singleton (bag.emtpy String)) = false
```

File: src/theory/bags/normal_form.cpp [hunk 7]
```diff
@@ -600,11 +600,12 @@ Node NormalForm::evaluateIsSingleton(TNode n)
...
-  // - (bag.is_singleton (MK_BAG "x" 1)) = true
-  // - (bag.is_singleton (MK_BAG "x" 4)) = false
-  // - (bag.is_singleton (union_disjoint (MK_BAG "x" 1) (MK_BAG "y" 1))) = false
+  // - (bag.is_singleton (BAG_MAKE "x" 1)) = true
+  // - (bag.is_singleton (BAG_MAKE "x" 4)) = false
+  // - (bag.is_singleton (union_disjoint (BAG_MAKE "x" 1) (BAG_MAKE "y" 1))) =
```
Comment (aniemetz) [inline src/theory/bags/normal_form.cpp:605]: Ditto

File: src/theory/bags/normal_form.h [hunk 8]
```diff
@@ -30,11 +30,11 @@ class NormalForm
   /**
    * Returns true if n is considered a to be a (canonical) constant bag value.
    * A canonical bag value is one whose AST is:
-   *   (union_disjoint (mkBag e1 c1) ...
-   *      (union_disjoint (mkBag e_{n-1} c_{n-1}) (mkBag e_n c_n))))
+   *   (union_disjoint (bag e1 c1) ...
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:33]:
```suggestion
   *   (bag.union_disjoint (bag e1 c1) ...
```

File: src/theory/bags/normal_form.h [hunk 9]
```diff
@@ -30,11 +30,11 @@ class NormalForm
...
+   *      (union_disjoint (bag e_{n-1} c_{n-1}) (bag e_n c_n))))
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:34]:
```suggestion
   *      (bag.union_disjoint (bag e_{n-1} c_{n-1}) (bag e_n c_n))))
```

File: src/theory/bags/normal_form.h [hunk 10]
```diff
@@ -112,8 +112,8 @@ class NormalForm
                                       T5&& remainderOfB);
   /**
    * evaluate n as follows:
-   * - (mkBag a 0) = (emptybag T) where T is the type of the original bag
-   * - (mkBag a (-c)) = (emptybag T) where T is the type the original bag,
+   * - (bag a 0) = (emptybag T) where T is the type of the original bag
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:115]:
```suggestion
   * - (bag a 0) = (bag.empty T) where T is the type of the original bag
```

File: src/theory/bags/normal_form.h [hunk 11]
```diff
@@ -112,8 +112,8 @@ class NormalForm
...
+   * - (bag a (-c)) = (emptybag T) where T is the type the original bag,
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:116]:
```suggestion
   * - (bag a (-c)) = (bag.empty T) where T is the type the original bag,
```

File: src/theory/bags/normal_form.h [hunk 12]
```diff
@@ -135,8 +135,8 @@ class NormalForm
   /**
    * evaluates union disjoint node such that the returned node is a canonical
    * bag that has the form
-   * (union_disjoint (mkBag e1 c1) ...
-   *   (union_disjoint  * (mkBag e_{n-1} c_{n-1}) (mkBag e_n c_n)))) where
+   * (union_disjoint (bag e1 c1) ...
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:138]:
```suggestion
   * (bag.union_disjoint (bag e1 c1) ...
```

File: src/theory/bags/normal_form.h [hunk 13]
```diff
@@ -135,8 +135,8 @@ class NormalForm
...
+   *   (union_disjoint  * (bag e_{n-1} c_{n-1}) (bag e_n c_n)))) where
```
Comment (aniemetz) [inline src/theory/bags/normal_form.h:139]:
```suggestion
   *   (bag.union_disjoint  * (bag e_{n-1} c_{n-1}) (bag e_n c_n)))) where
```

File: src/api/cpp/cvc5.cpp [hunk 14]
```diff
@@ -295,16 +295,16 @@ const static std::unordered_map<Kind, cvc5::Kind> s_kinds{
...
     /* Bags ---------------------------------------------------------------- */
-    {UNION_MAX, cvc5::Kind::UNION_MAX},
-    {UNION_DISJOINT, cvc5::Kind::UNION_DISJOINT},
-    {INTERSECTION_MIN, cvc5::Kind::INTERSECTION_MIN},
-    {DIFFERENCE_SUBTRACT, cvc5::Kind::DIFFERENCE_SUBTRACT},
-    {DIFFERENCE_REMOVE, cvc5::Kind::DIFFERENCE_REMOVE},
-    {SUBBAG, cvc5::Kind::SUBBAG},
+    {BAG_UNION_MAX, cvc5::Kind::BAG_UNION_MAX},
+    {BAG_UNION_DISJOINT, cvc5::Kind::BAG_UNION_DISJOINT},
+    {BAG_INTERSECTION_MIN, cvc5::Kind::BAG_INTERSECTION_MIN},
```
Comment (aniemetz) [inline src/api/cpp/cvc5.cpp:300]: We should call this kind `BAG_INTER_MIN` to be consistent with sets and regexp.

Comment (mudathirmahgoub) [reply src/api/cpp/cvc5.cpp]: Done. I thought only the smtlib needs to be changed before.

File: src/theory/bags/bag_make_op.h [hunk 15]
```diff
@@ -25,39 +25,39 @@ namespace cvc5 {
 class TypeNode;
 
 /**
- * The class is an operator for kind MK_BAG used to construct bags.
+ * The class is an operator for kind BAG_MAKE used to construct bags.
  * It specifies the type of the element especially when it is a constant.
  * e.g. the type of rational 1 is Int, however
- * (mkBag (mkBag_op Real) 1) is of type (Bag Real), not (Bag Int).
+ * (bagMake (mkBag_op Real) 1) is of type (Bag Real), not (Bag Int).
```
Comment (aniemetz) [inline src/theory/bags/bag_make_op.h:31]:
```suggestion
 * (bagMake (BagMakeOp Real) 1) is of type (Bag Real), not (Bag Int).
```

File: src/theory/bags/bags_rewriter.cpp [hunk 16]
```diff
@@ -467,7 +470,7 @@ BagsRewriteResponse BagsRewriter::rewriteFromSet(const TNode& n) const
   Assert(n.getKind() == BAG_FROM_SET);
   if (n[0].getKind() == SET_SINGLETON)
   {
-    // (bag.from_set (set.singleton (singleton_op Int) x)) = (mkBag x 1)
+    // (bag.from_set (set.singleton (singleton_op Int) x)) = (bag x 1)
```
Comment (aniemetz) [inline src/theory/bags/bags_rewriter.cpp:473]:
```suggestion
    // (bag.from_set (set.singleton (SetSingletonOp Int) x)) = (bag x 1)
```

File: src/theory/bags/bags_rewriter.cpp [hunk 17]
```diff
@@ -478,10 +481,10 @@ BagsRewriteResponse BagsRewriter::rewriteFromSet(const TNode& n) const
 BagsRewriteResponse BagsRewriter::rewriteToSet(const TNode& n) const
 {
   Assert(n.getKind() == BAG_TO_SET);
-  if (n[0].getKind() == MK_BAG && n[0][1].isConst()
+  if (n[0].getKind() == BAG_MAKE && n[0][1].isConst()
       && n[0][1].getConst<Rational>().sgn() == 1)
   {
-    // (bag.to_set (mkBag x n)) = (singleton (singleton_op T) x)
+    // (bag.to_set (bag x n)) = (set.singleton (singleton_op T) x)
```
Comment (aniemetz) [inline src/theory/bags/bags_rewriter.cpp:487]:
```suggestion
    // (bag.to_set (bag x n)) = (set.singleton (SetSingletonOp T) x)
```

File: src/theory/bags/bags_rewriter.h [hunk 18]
```diff
@@ -142,32 +142,32 @@ class BagsRewriter : public TheoryRewriter
...
-   * - (difference_subtract A (union_max A B)) = emptybag
-   * - (difference_subtract A (union_max B A)) = emptybag
-   * - (difference_subtract (intersection_min A B) A) = emptybag
-   * - (difference_subtract (intersection_min B A) A) = emptybag
+   * - (difference_subtract A (union_disjoint A B)) = bag.empty
+   * - (difference_subtract A (union_disjoint B A)) = bag.empty
+   * - (difference_subtract A (union_max A B)) = bag.empty
+   * - (difference_subtract A (union_max B A)) = bag.empty
+   * - (difference_subtract (intersection_min A B) A) = bag.empty
+   * - (difference_subtract (intersection_min B A) A) = bag.empty
```
Comment (aniemetz) [inline src/theory/bags/bags_rewriter.h:155]: Missing `bag.` prefix, and `intersection_min` -> `bag.inter_min`. There's more throughout the whole file, make sure to rename all of them.

File: src/theory/bags/normal_form.h (hunk 8 above)
Comment (aniemetz) [reply src/theory/bags/normal_form.h:33]: You missed this one.

File: src/theory/bags/normal_form.h (hunk 9 above)
Comment (aniemetz) [reply src/theory/bags/normal_form.h:34]: And this one.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**